source/shadow.c

```diff
@@ -297,6 +398,41 @@ static ShadowStatus_t validateName( const char * pString,
 
 /*-----------------------------------------------------------*/
 
+static ShadowStatus_t extractThingName( const char * pTopic,
+                                        uint16_t topicLength,
+                                        uint16_t * pConsumedTopicLength,
+                                        uint16_t * pThingNameLength )
+{
+    ShadowStatus_t shadowStatus;
+
+    /* Extract thing name. */
+    shadowStatus = validateName( &( pTopic[ *pConsumedTopicLength ] ),
+                                 topicLength - *pConsumedTopicLength,
+                                 pThingNameLength );
+
+    if( shadowStatus != SHADOW_SUCCESS )
+    {
+        shadowStatus = SHADOW_THINGNAME_PARSE_FAILED;
+        LogDebug( ( "Not related to Shadow, failed to parse thing name in pTopic %s.", pTopic ) );
```

Comment:
Might be clearer to just use comma or period after "Not related to Shadow" in logs